CMakeLists.txt

```diff
@@ -1,74 +1,56 @@
-# CMake project file for FOO
-
-##################################################
-# Define the project and the depencies that it has
-##################################################
-
-CMAKE_MINIMUM_REQUIRED(VERSION 2.8.5)
-PROJECT(FOO Fortran)
-
-# Set the FOO version
-SET(VERSION 1.2.3)
-
-# Add our local modlues to the module path
-SET(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake/Modules/")
-
-# Uncomment if it is required that Fortran 90 is supported
-#IF(NOT CMAKE_Fortran_COMPILER_SUPPORTS_F90)
-#    MESSAGE(FATAL_ERROR "Fortran compiler does not support F90")
-#ENDIF(NOT CMAKE_Fortran_COMPILER_SUPPORTS_F90)
-
-# Set some options the user may choose
-# Uncomment the below if you want the user to choose a parallelization library
-#OPTION(USE_MPI "Use the MPI library for parallelization" OFF)
-#OPTION(USE_OPENMP "Use OpenMP for parallelization" OFF)
-
-# This INCLUDE statement executes code that sets the compile flags for DEBUG,
-# RELEASE, and TESTING.  You should  review this file and make sure the flags 
-# are to your liking.
-INCLUDE(${CMAKE_MODULE_PATH}/SetFortranFlags.cmake) 
-# Locate and set parallelization libraries.  There are some CMake peculiarities
-# taken care of here, such as the fact that the FindOpenMP routine doesn't know
-# about Fortran.
-#INCLUDE(${CMAKE_MODULE_PATH}/SetParallelizationLibrary.cmake)
-# Setup the LAPACK libraries.  This also takes care of peculiarities, such as
-# the fact the searching for MKL requires a C compiler, and that the results
-# are not stored in the cache. 
-#INCLUDE(${CMAKE_MODULE_PATH}/SetUpLAPACK.cmake)
-
-# There is an error in CMAKE with this flag for pgf90.  Unset it
-GET_FILENAME_COMPONENT(FCNAME ${CMAKE_Fortran_COMPILER} NAME)
-IF(FCNAME STREQUAL "pgf90")
-    UNSET(CMAKE_SHARED_LIBRARY_LINK_Fortran_FLAGS)
-ENDIF(FCNAME STREQUAL "pgf90")
-
-############################################################
-# Define the actual files and folders that make up the build
-############################################################
-
-# Define the executable name
-SET(FOOEXE foo)
-
-# Define the library name
-SET(BARLIB bar)
-
-# Define some directories
-SET(SRC ${CMAKE_SOURCE_DIR}/src)
-SET(LIB ${CMAKE_SOURCE_DIR}/lib)
-SET(BIN ${CMAKE_SOURCE_DIR}/bin)
-SET(SRCFOO ${SRC}/foo)
-SET(SRCBAR ${SRC}/bar)
-
-# Have the .mod files placed in the lib folder
-SET(CMAKE_Fortran_MODULE_DIRECTORY ${LIB})
-
-# The source for the BAR library and have it placed in the lib folder
-ADD_SUBDIRECTORY(${SRCBAR} ${LIB})
-
-# The source for the FOO binary and have it placed in the bin folder
-ADD_SUBDIRECTORY(${SRCFOO} ${BIN})
-
-# Add a distclean target to the Makefile
-ADD_CUSTOM_TARGET(distclean 
-    COMMAND ${CMAKE_COMMAND} -P ${CMAKE_SOURCE_DIR}/distclean.cmake
-)
+cmake_minimum_required(VERSION 3.15.2)
```

Comment:
Is 3.15.2 actually the minimum version? At the time of this writing I think that is pretty new, and perhaps if a slightly older version is supported by this template we should allow that.

This is from someone who works in an environment where often only older software versions are supported, so I know first hand the pain when you cannot use something shiny because your tools are too old.

Reply:
I only have 3.15.2 installed on the machine I developed and tested this with.  It is possible that I could get an older version and try again to see if it builds.  I suspect it may work back to 3.11 or so, but I believe in some cases I am using newer features or syntax that will limit the ability to backport and still call it "modern" CMake.

Let me try an older version and get back to you.

Reply:
`target_link_options` needs at least 3.13, so I don't think 3.11 will work (without changing the code).